src/client/components/series-menu/format-picker.tsx

```diff
@@ -42,24 +42,24 @@ function readFormat(format: string, measureFormat: string): SeriesFormat {
   }
 }
 
-function printFormat(format: SeriesFormat, measureFormat: string): string {
+function printFormat(format: SeriesFormat, measureFormat: {} | string): string {
   switch (format.type) {
     case SeriesFormatType.DEFAULT:
-      return measureFormat;
+      return typeof measureFormat === "string" ? measureFormat : JSON.stringify(measureFormat);
     case SeriesFormatType.EXACT:
       return exactFormat;
     case SeriesFormatType.PERCENT:
       return percentFormat;
     case SeriesFormatType.CUSTOM:
-      return format.value;
+      return typeof format.value === "string" ? format.value : JSON.stringify(format.value);
```

Comment:
Why we stringily custom format? Do we allow user to insert json object in input? If so, we should handle reading in `readFormat`